aya/src/programs/uprobe.rs

```diff
@@ -121,6 +122,50 @@ impl UProbe {
 
         attach(&mut self.data, self.kind, &path, sym_offset + offset, pid)
     }
+
+    /// Attaches the program to an address within userspace process' own
+    /// address space.
+    ///
+    /// Attaches the uprobe to the given address defined in the `addr`
+    /// argument. If `pid` is not `None`, the program executes only when the target
+    /// function is executed by the given `pid`.
+    ///
+    /// If the program is an `uprobe`, it is attached to the *start* address of the target
+    /// function.  Instead if the program is an `uretprobe`, it is attached to the return address of
+    /// the target function.
```

Comment:
This should clearly document the fact that it relies on the function to _not_ be inlined.

Reply:
Done